### ISSUE TITLE
List of official devel/unstable branches

### ISSUE THREAD

**Issue by AMDmi3** [repology/repology#316]
Let's collect a list of devel/unstable software branches, so when #170 is implemented it can be used to make stable branches not counted as outdated. We need

- version pattern
- upstream url which could be used to keep the pattern up to date

| Project | Pattern | URL / comments |
|-|-|-|
| apr | `2\\..*` | http://apr.apache.org/ |
| gcc | `[0-9]+\\.0.*`, `[0-9]+\\.[0-9]+\\.[1-9]` | https://gcc.gnu.org/releases.html, https://mirrors.kernel.org/sourceware/gcc/releases/ |
| GNOME<sup>1</sup> | `[0-9]+\\.[0-9]*[13579]\\..*` | https://developer.gnome.org/programming-guidelines/stable/versioning.html.en#stable-unstable-versionsneed |
| less | >487<sup>2</sup> | http://www.greenwoodsoftware.com/less/ |
| libressl | `2\\.6\\..*` | https://www.libressl.org/ |
| postgresql | `10\\..*` | https://www.postgresql.org/developer/beta/ |
| tor | `3\\.1\\..*` | https://www.torproject.org/download/download.html.en |
| wine | `[0-9]+\\.[1-9].*` | https://www.winehq.org/download |

- slashes are doubled to simplify pasting into rules .yaml
- <sup>1</sup> need full list of metapackages
- <sup>2</sup> need new kind of version comparison

**Comment by jrmarino** [comment]
some gnome metapackages affected:

- adwaita-icon-theme
- at-spi2-atk
- at-spi2-core
- atk
- cairo
- glib
- gobject-introspection

**Comment by vcunat** [comment]
There could also be some general rules, e.g. checks for substrings like `rc`, `beta`, ...

**Comment by AMDmi3** [comment]
Sure, that's also planned

**Comment by AMDmi3** [comment]
Listed packages should be fixed. Global rule for all rc/alpha/beta versions is not there yet

**Comment by AMDmi3** [comment]
It's there now. Not for `[ab][0-9]+$` yet though, but may add as well.

**Comment by jtojnar** [comment]
Some GNOME packages listing unstable version (3.27):

* gnome-disk-utility
* gnome-keyring
* gnome-maps
* gnome-mines
* gnome-online-accounts
* gnome-photos
* gnome-shell
* gnome-shell-extensions
* gnome-software
* gnome-tweak-tool

**Comment by AMDmi3** [comment]
I've added these and some more rules. It's easier to get them raw from the database, using `select distinct effname from packages where version like '3.27.%' and devel=false;` query.

**Comment by jtojnar** [comment]
`webkitgtk` also uses the “odd minor unstable” rule.

**Comment by jtojnar** [comment]
Also these:

* eog
* geocode-glib
* gnome-boxes
* geocode-glib
* atkmm

**Comment by jtojnar** [comment]
Also `python:pycairo`

**Comment by jtojnar** [comment]
Two more `empathy` and `gnome-sudoku`.

**Comment by jtojnar** [comment]
Also `gcr` and `dconf-editor`.

**Comment by jtojnar** [comment]
Also `gdm` and `gnome-music`.

**Comment by AMDmi3** [comment]
Would be nice if you create new issues for future reports, as there are chances I may miss followups to this closed issue. Thank you!

**Comment by AMDmi3** [comment]
Also note to self: can get the list of more candidates for gnome devel versioning scheme with this query:

```
select effname,version from packages where version like '3.27.%' and versionclass = 1;
```

**Comment by vcunat** [comment]
bind newly switched to the scheme where even=stable and odd=unstable. https://lists.isc.org/pipermail/bind-announce/2018-May/001092.html

**Comment by AMDmi3** [comment]
Yeah, I've seen the news, adding rule for this was on my list. Thanks for the report anyway!

**Comment by AMDmi3** [comment]
Fixed by repology/repology-rules#33
